tests/auth.js

```diff
@@ -0,0 +1,143 @@
+      st.kill()
+      return true
+    })
+    st.stdout.empty()
+    st.fails()
+    st.end()
+  })
+
+  test('auth - register works', function (t) {
+    // cmd: dat clone <link> (same thing again as above)
+    var cmd = dat + ' register --email=hello@bob.com --password=joe --server=' + SERVER
+    var st = spawn(t, cmd, {cwd: baseTestDir})
+    st.stdout.match(function (output) {
+      t.same(output.trim(), 'Registered successfully.', 'output success message')
+      return true
+    })
+    st.stderr.empty()
+    st.end()
+  })
+
+  test('auth - login requires email', function (t) {
+    // cmd: dat clone <link> (same thing again as above)
+    var cmd = dat + ' login --password=joe --server=' + SERVER
+    var st = spawn(t, cmd, {cwd: baseTestDir})
+    st.stderr.match(function (output) {
+      t.ok(output.indexOf('Email is required') > -1, 'outputs correct error message')
+      st.kill()
+      return true
+    })
+    st.stdout.empty()
+    st.end()
+  })
+
+  test('auth - login requires email and password', function (t) {
+    // cmd: dat clone <link> (same thing again as above)
+    var cmd = dat + ' login --email=hello@bob.com --server=' + SERVER
+    var st = spawn(t, cmd, {cwd: baseTestDir})
+    st.stderr.match(function (output) {
+      t.ok(output.indexOf('Password is required') > -1, 'outputs correct error message')
+      st.kill()
+      return true
+    })
+    st.stdout.empty()
+    st.end()
+  })
+
+  test('auth - login works', function (t) {
+    // cmd: dat clone <link> (same thing again as above)
+    var cmd = dat + ' login --email=hello@bob.com --password=joe --server=' + SERVER
+    var st = spawn(t, cmd, {cwd: baseTestDir})
+    st.stdout.match(function (output) {
+      t.same(output.trim(), 'Logged in successfully.', 'output success message')
+      return true
+    })
+    st.stderr.empty()
+    st.end()
+  })
+
+  test('auth - whoami works', function (t) {
+    // cmd: dat clone <link> (same thing again as above)
+    var cmd = dat + ' whoami --server=' + SERVER
+    var st = spawn(t, cmd, {cwd: baseTestDir})
+    st.stdout.match(function (output) {
+      t.same('hello@bob.com', output, 'email printed')
+      return true
+    })
+    st.stderr.empty()
+    st.end()
+  })
+
+  test('auth - logout works', function (t) {
+    // cmd: dat clone <link> (same thing again as above)
+    var cmd = dat + ' logout'
+    var st = spawn(t, cmd, {cwd: baseTestDir})
+    st.stdout.match(function (output) {
+      t.same('Logged out.', output, 'output correct')
+      return true
+    })
+    st.stderr.empty()
+    st.end()
+  })
+
+  test('auth - logout prints correctly when trying to log out twice', function (t) {
+    // cmd: dat clone <link> (same thing again as above)
+    var cmd = dat + ' logout'
+    var st = spawn(t, cmd, {cwd: baseTestDir})
+    st.stdout.match(function (output) {
+      t.same('Not logged in.', output, 'output correct')
+      return true
+    })
+    st.stderr.empty()
+    st.end()
+  })
+
```

Comment:
We should add a test making sure whoami is empty after logout. Once ready, we'll need to have a secure request test after logout so that can be a placeholder.